otherlibs/dynlink/Makefile

```diff
@@ -189,10 +189,12 @@ $(LOCAL_SRC)/%.cmx: $(LOCAL_SRC)/%.ml
 # Rules for building the [Dynlink_compilerlibs] bytecode and native packs
 # from their components.
 
-byte/dynlink_compilerlibs.cmo: $(COMPILERLIBS_CMO)
-	$(OCAMLC) $(COMPFLAGS) -pack -o $@ $(COMPILERLIBS_CMO)
-
-byte/dynlink_compilerlibs.cmi: byte/dynlink_compilerlibs.cmo
+# A multi-target pattern rule is used here as grouped targets require
+# GNU make 4.3+.
+byt%/dynlink_compilerlibs.cmo byt%/dynlink_compilerlibs.cmi: $(COMPILERLIBS_CMO)
+	@$(if $(filter-out e,$*),\
+        $(error Should only build byte/dynlink_compilerlibs.cmo!))
```

Comment:
This slightly pedantic guard codifies the fact that we only expect this pattern to be used with `$*` = `e`. It means `make -C otherlibs/dynlink bytx/dynlink_compilerlibs.cmo` (or any other typo) produces a clear error message rather than _silently_ rebuilding `byte/dynlink_compilerlibs.cmo` and `byte/dynlink_compilerlibs.cmi`